salt/metalk8s/orchestrate/deploy_new_node.sls

```diff
@@ -35,13 +31,22 @@ Accept key:
     - require:
       - salt: Deploy salt-minion on a new node
 
+Wait minion available:
+  salt.runner:
+    - name: metalk8s_saltutil.wait_minions
+    - tgt: {{ pillar['node_name'] }}
+    - require:
+      - salt: Accept key
+    - require_in:
+      - salt: Set grains
+{%- endif %}
+
 Set grains:
   salt.state:
     - tgt: {{ pillar['node_name'] }}
+    - saltenv: {{ saltenv }}
```

Comment:
This would be the `saltenv` with which the runner is invoked. However, that's not necessarily the `saltenv` of what we want to deploy on the node: that's based on its `metalk8s.scality.com/version` annotation, so the `saltenv` is `metalk8s-{{ pillar.metalk8s.nodes[pillar['node_name']].version }}` (or something along those lines).

Reply:
Oh yes indeed I will update